LBANN cannot be built with "make"

Nobody should build LBANN with `make` anyway (`pip install ninja` or however you might want to acquire it), but if someone were to try to use the `Unix Makefiles` generator with CMake, they would get a message similar to the following:
```
> make
[  0%] Running protoc on LBANN protobuf sources.
/path/to/lbann/build/src/proto/lbann/proto/: No such file or directory
make[2]: *** [src/proto/CMakeFiles/LbannProto_genSrc.dir/build.make:117: src/proto/lbann/proto/lbann.pb.cc] Error 1
make[1]: *** [CMakeFiles/Makefile2:3308: src/proto/CMakeFiles/LbannProto_genSrc.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/src/proto/CMakeLists.txt
+++ b/src/proto/CMakeLists.txt
@@ -58,6 +58,7 @@ if (LBANN_HAS_PYTHON_FRONTEND)
 endif (LBANN_HAS_PYTHON_FRONTEND)
   endforeach ()
   add_custom_command(
+    COMMAND ${CMAKE_COMMAND} -E make_directory "${CMAKE_CURRENT_BINARY_DIR}/lbann/proto"
     COMMAND protobuf::protoc
     "--cpp_out=${CMAKE_CURRENT_BINARY_DIR}/lbann/proto"
     $<$<BOOL:${LBANN_HAS_PYTHON_FRONTEND}>:--python_out=${CMAKE_CURRENT_BINARY_DIR}>

